presto-iceberg/src/main/java/com/facebook/presto/iceberg/TableStatisticsMaker.java

```diff
@@ -497,8 +527,38 @@ private Optional<StatisticsFile> getClosestStatisticsFileForSnapshot(IcebergTabl
     /**
      * Builds a map of field ID to ColumnStatistics for a particular {@link StatisticsFile}.
      */
-    private Map<Integer, ColumnStatistics.Builder> loadStatisticsFile(StatisticsFile file)
+    private Map<Integer, ColumnStatistics> loadStatisticsFile(IcebergTableHandle tableHandle, StatisticsFile file, StatisticsFileCache statisticsFileCache, List<Integer> columnIds)
     {
+        // first, try to load all stats from the cache. If the map doesn't contain all keys, load the missing
+        // stats into the cache.
+        Map<Integer, ColumnStatistics> cachedStats = columnIds.stream()
+                .map(id -> Pair.of(id, statisticsFileCache.getIfPresent(new StatisticsFileCacheKey(file, id))))
+                .filter(pair -> pair.second() != null)
+                .collect(toImmutableMap(Pair::first, Pair::second));
+        Set<Integer> missingStats = columnIds.stream().filter(id -> !cachedStats.containsKey(id)).collect(toImmutableSet());
+        if (missingStats.isEmpty()) {
+            return cachedStats;
+        }
+        if (!cachedStats.isEmpty()) {
+            session.getRuntimeStats().addMetricValue(
+                    String.format(STATISITCS_CACHE_METRIC_PARTIAL_MISS_FORMAT,
+                            tableHandle.getSchemaTableName(),
+                            murmur3_128().hashString(file.path(), UTF_8)),
+                    RuntimeUnit.NANO, 1);
+        }
+
+        String fileSizeMetricName = String.format(STATISITCS_CACHE_METRIC_FILE_SIZE_FORMAT,
+                tableHandle.getSchemaTableName(),
+                murmur3_128().hashString(file.path(), UTF_8));
```

Comment:
Isn't the file path unique for every stats file ? Also, does it always include the table name ? If so, we can skip hashing the `file.path()` and also skip the table name for this metric

Reply:
The stats files are always unique, but it is not guaranteed that the path always includes the table name. People can configure the storage directory however they like which means it may or may not include the table name. Generally with default settings, the table names exist in the path though

Reply:
Okay lets include the table name in the metric :+1: 
My reasoning behind having the full file path in the metric name was to help users identify which stats files are large in size. Since the path is unique, should we skip hashing it altogether ?

Reply:
Updated to remove the hash